sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc

```diff
@@ -195,11 +195,11 @@ class device_global {
 
 `device_global` is a class template, parameterized by the type of the underlying allocation _T_, and a list of properties _PropertyListT_. The type of the allocation _T_ also encodes the size of the allocation for potentially multidimensional array types.
 
-_T_ is restricted to types that have a trivial destructor and a trivial default constructor in this revision of the specification (the constructor restriction may be partially relaxed in a future revision). _PropertyListT_ enables properties to be associated with a `device_global`.
+_T_ is restricted to types that have a trivial destructor. Furthermore, `T` must also have a trivial default constructor when compiling with {cpp} versions before {cpp}20. _PropertyListT_ enables properties to be associated with a `device_global`.
 
-Since _T_ is restricted to types with trivial default constructors in this version of the specification, there are no non-default `device_global` constructors, and therefore no initialization values may be specified for the content of a `device_global` allocation on a device.
+If no trivial default constructor is required with _T_, a non-default  `consteval` `device_global` constructor can be called. This allows for the specification of initialization values for the contents of a `device_global` allocation. The allocation will occur before the first access to that `device_global` on that device.
 
-The allocation of type _T_ for a given `device_global` is zero-initialized on a given device prior to the first access to that `device_global` on that device. For the purposes of this definition an access can be a direct access of the `device_global` in kernel code or a copy to or from that `device_global` enqueued to the given device.
+If a trivial default constructor is required, or the default `device_global` constructor has been called, the allocation of type _T_ for a given `device_global` is zero-initialized on a given device prior to the first access to that `device_global` on that device. For the purposes of this definition an access can be a direct access of the `device_global` in kernel code or a copy to or from that `device_global` enqueued to the given device.
```

Comment:
I find these paragraphs really confusing.  Can we say it like this:

> _T_ is restricted to types that have a trivial destructor.  _PropertyListT_ enables properties to be associated with a `device_global`.
>
> When compiling with C++ versions before C++20, _T_ must also have a trivial default constructor.  In this case, the allocation of type _T_ for a given `device_global` is zero-initialized on a given device prior to the first access to that `device_global` on that device. For the purposes of this definition an access can be a direct access of the `device_global` in kernel code or a copy to or from that `device_global` enqueued to the given device.
>
> When compiling with C++20 or later, _T_ must have a constructor that can be `constexpr` evaluated, and the parameters to the `device_global` constructor are forwarded to the _T_ constructor.  In this case, the allocation of type _T_ for a given `device_global` is initialized on a given device prior to the first access to that `device_global` on that device.

Reply:
I updated the spec to include these changes.